### ISSUE TITLE
Allow symbols in attrib dicts?

### ISSUE THREAD

**Issue by adigitoleo** [Alexander-Barth/NCDatasets.jl#154]
It's not really a bug, but the current implementation does not allow for `Symbol` types in the attribute dictionaries, e.g.:

```julia
julia> var = OrderedDict(:long_name => "variable")
OrderedDict{Symbol, String} with 1 entry:
  :long_name => "variable"

julia> NCDataset("/tmp/test.nc", "c") do dataset
           defDim(dataset, "lon", 100)
           defDim(dataset, "lat", 110)
           defVar(dataset, "var", String, ("lon", "lat"), attrib = var)
       end
ERROR: MethodError: no method matching setindex!(::NCDatasets.Attributes{NCDataset{Nothing}}, ::String, ::Symbol)
Closest candidates are:
  setindex!(::NCDatasets.Attributes, ::Any, ::AbstractString) at ~/vcs/NCDatasets.jl/src/attributes.jl:72
Stacktrace:
 [1] defVar(ds::NCDataset{Nothing}, name::String, vtype::DataType, dimnames::Tuple{String, String}; kwargs::Base.Pairs{Symbol, OrderedDict{Symbol, String}, Tuple{Symbol}, NamedTuple{(:attrib,), Tuple{OrderedDict{Symbol, String}}}})
   @ NCDatasets ~/vcs/NCDatasets.jl/src/cfvariable.jl:144
 [2] (::var"#1#2")(dataset::NCDataset{Nothing})
   @ Main ./REPL[7]:4
 [3] NCDataset(::var"#1#2", ::String, ::Vararg{String}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ NCDatasets ~/vcs/NCDatasets.jl/src/dataset.jl:220
 [4] NCDataset(::Function, ::String, ::Vararg{String})
   @ NCDatasets ~/vcs/NCDatasets.jl/src/dataset.jl:218
 [5] top-level scope
   @ REPL[7]:1
```

It surprised me, since symbols are allowed in `OrderedDict`s. I wonder if the type constraint on `setindex` and any other affected methods could be changed to `Union{AbstractString, Symbol}` without negative effects? Alternatively, perhaps this case deserves a special error message, something like "Symbols are not allowed in attribute dictionaries"?

Low priority, I'm willing to take this on if you don't have time.

**Comment by Alexander-Barth** [comment]
I agree that this would be a nice feature. Similarly symbols could also be used for dimension and group names.

**Comment by Alexander-Barth** [comment]
As you have seen, I merged your PR (thanks again!). If you have time to check if all works as intended on your side, I will make a new release.

**Comment by adigitoleo** [comment]
I botched it, sorry. PR to fix it #157 .
